search/query-features-with-arcade-expression/src/main/java/com/esri/samples/query_features_with_arcade_expression/QueryFeaturesWithArcadeExpressionSample.java

```diff
@@ -0,0 +1,198 @@
+
+      // create a map view and set the map to it
+      mapView = new MapView();
+      mapView.setMap(map);
+
+      // set callout properties that never change
+      mapView.getCallout().setLeaderPosition(Callout.LeaderPosition.BOTTOM);
+      mapView.getCallout().setTitle("RPD Beats");
+
+      // wait for the map to finish loading data from portal before accessing its layers
+      map.addDoneLoadingListener(() -> {
+        if (map.getLoadStatus() == LoadStatus.LOADED) {
+          // hide all layers other than the RPD Beats Layer to reduce clutter
+          map.getOperationalLayers().forEach(layer -> layer.setVisible(layer.getName().equals(rpdBeatsLayerName)));
+
+          // get the RPD beats layer.
+          policeBeatsLayer = map.getOperationalLayers().stream().filter(
+            layer -> layer.getName().equals(rpdBeatsLayerName)).toList().get(0);
+
+          // if map clicked, evaluate an arcade expression at that point
+          mapView.addEventHandler(MouseEvent.MOUSE_CLICKED, event -> {
+            if (event.getButton() == MouseButton.PRIMARY && event.isStillSincePress()) {
+              try {
+                evaluateArcadeExpression(new Point2D(event.getX(), event.getY()));
+              } catch (Exception e) {
+                throw new RuntimeException(e);
+              }
+            }});
+        } else if (map.getLoadStatus() == LoadStatus.FAILED_TO_LOAD) {
+          new Alert(Alert.AlertType.ERROR, "Map failed to load").show();
+        }
+      });
+
+      // add the map view to the stack pane
+      stackPane.getChildren().addAll(mapView);
+    } catch (Exception e) {
+      // on any error, display the stack trace.
+      e.printStackTrace();
+    }
+  }
+
+  /**
+   * Evaluates an arcade expression at the point on the map that was clicked.
+   *
+   * @param point The point on the map which was clicked, in screen (rather than map) coordinates.
+   * @throws RuntimeException If unable to get results from futures due to interruption or concurrent execution.
+   */
+  private void evaluateArcadeExpression(Point2D point) throws RuntimeException {
+    // convert the passed point from screen coordinates to map coordinates, for use in displaying the callout
+    Point mapPoint = mapView.screenToLocation(point);
+
+    var identifyLayerResultFuture = mapView.identifyLayerAsync(policeBeatsLayer, point,
+      0, false);
+    identifyLayerResultFuture.addDoneListener(() -> {
+      if (identifyLayerResultFuture.isDone()) {
+        try {
+          var identifyLayerResult = identifyLayerResultFuture.get();
+
+          // only execute query if the police beats layer has data at the selected point
+          if (identifyLayerResult.getElements().size() != 0) {
+            ArcGISFeature feature = (ArcGISFeature) identifyLayerResult.getElements().get(0);
+
+            // show callout and text without data
+            mapView.getCallout().setTitle(calloutText);
+            mapView.getCallout().setDetail("");
```

Comment:
can we set the calloutText as the detail, and keep the Title as it was set at the beginning of the sample? Whilst there is a slight delay, I suggest something like the following to indicate some populating is happening. I originally tried with "Calculating" but the time lag is so minimal that it just looked weird seeing the callout go from a long to a shorter width as it adapted to the longer "Calculating" word. Maybe you could set a min size to the callout to counter that if you preferred that to the "..." approach.

```java
            mapView.getCallout().setDetail("..." + calloutText);

```